src/Authentication/Authenticators/AccessTokens.php

```diff
@@ -154,6 +154,19 @@ public function check(array $credentials): Result
 
         assert($token->last_used_at instanceof Time || $token->last_used_at === null);
 
+        // Is expired ?
+        if (
+            $token->expires !== null
```

Comment:
can this use instanceof, eg : 

```suggestion
            $token->expires instanceof Time
```

Reply:
Sounds good.

Reply:
@CosDiabos Would you find time for these final changes? I'm looking forward to merging this PR. Thanks!